frontend/packages/design-system/src/types/icons.ts

```diff
@@ -1,60 +1,7 @@
-export type IconProps = {
-  name: IconName;
-  variant: IconVariants;
-};
+import { IIconProps } from 'native-base';
 
-export enum IconVariants {
-  DARK = '#000000',
-  LIGHT = '#FFFFFF',
-  LINK = '#1E4A7D',
-}
+import { iconMap } from '../assets/iconMap';
 
-export enum IconName {
-  ATTACHMENT = 'attachment',
-  BENEFICIARY = 'beneficiary',
-  CALENDAR = 'calendar',
-  CALL = 'call',
-  CASE = 'case',
-  CHAT = 'chat',
-  DELETE = 'delete',
-  EDIT = 'edit',
-  EMAIL = 'email',
-  FACE_ERROR = 'face_error',
-  FACE_GENERAL = 'face_general',
-  FACE_SUCCESS = 'face_success',
-  FAVORITE = 'favorite',
-  FEMALE = 'female',
-  FILTER = 'filter',
-  HOME = 'home',
-  IMAGE = 'image',
-  LOCK = 'lock',
-  MALE = 'male',
-  MENU = 'menu',
-  MORE = 'more',
-  NEXT = 'next',
-  NOTIFICATION = 'notification',
-  PLUS = 'plus',
-  PRINT = 'print',
-  PROTECTION = 'protection',
-  REGISTRATION = 'registration',
-  RELOAD = 'reload',
-  SAVE = 'save',
-  SEARCH = 'search',
-  SETTING = 'setting',
-  SHARE = 'share',
-  SYNC = 'sync',
-  THUMBNAIL = 'thumbnail',
-  UNLOCK = 'unlock',
-  UPLOAD = 'upload',
-  USER_ADD = 'user_add',
-  USER_DELETE = 'user_delete',
-  USER_GROUP = 'user_group',
-  USER_GROUP_ADD = 'user_group_add',
-  WHATSAPP = 'whatsapp',
-  ACTIVITY_CAMP = 'activity_camp',
-  ACTIVITY_EDUCATION = 'activity_education',
-  ACTIVITY_FOOD = 'activity_food',
-  ACTIVITY_ICLA = 'activity_icla',
-  ACTIVITY_SHELTER = 'activity_shelter',
-  ACTIVITY_WASH = 'activity_wash',
-}
+export type IconProps = {
```

Comment:
I generally think props are best left next to their component, unless there's a very strong reason for sharing them?

Reply:
this is a small project (so far), but in general it happens a lot that types are used by multiple components or other types, and then there's the question of where to put them, and I don't like having some types with the components and some types somewhere else, so I like to have a types folder with all the types